src/PagePropsStatementCountLookup.php

```diff
@@ -14,13 +14,15 @@ public function __construct( LoadBalancer $loadBalancer ) {
 	}
 
 	public function getStatementCount( EntityId $entityId ) {
+		global $wgWBRepoSettings;
```

Comment:
I’m not sure I understand your concern. This extension already uses Wikibase classes (DataModel, Lib, and Repo) everywhere; surely it logically depends on Wikibase Repo, whether that’s encoded in the `composer.json` or not?

I guess a more object-oriented approach would be to pass an `EntityNamespaceLookup` to the `PagePropsStatementCountLookup` and call `getEntityNamespaces()` instead of looking at the settings, but that’s still a Wikibase class.